packages/loopring_v3/contracts/impl/owners/DelayedOwner.sol

```diff
@@ -0,0 +1,341 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../../lib/AddressUtil.sol";
+import "../../lib/BytesUtil.sol";
+import "../../lib/Claimable.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/ReentrancyGuard.sol";
+
+
+/// @title DelayedOwner
+/// @author Brecht Devos - <brecht@loopring.org>
+/// @dev Base class for an Owner contract where certain functions have
+///      a mandatory delay for security purposes.
+contract DelayedOwner is Claimable, ReentrancyGuard
+{
+    using AddressUtil for address payable;
+    using BytesUtil   for bytes;
+    using MathUint    for uint;
+
+    struct Transaction
+    {
+        uint    id;
+        uint    timestamp;
+        uint    value;
+        bytes   data;
+    }
+
+    struct DelayedFunction
+    {
+        bytes4  functionSelector;
+        uint    delay;
+    }
+
+    event TransactionDelayed(
+        uint    id,
+        uint    timestamp,
+        uint    value,
+        bytes   data,
+        uint    delay
+    );
+
+    event TransactionCancelled(
+        uint    id,
+        uint    timestamp,
+        uint    value,
+        bytes   data
+    );
+
+    event TransactionExecuted(
+        uint    timestamp,
+        uint    value,
+        bytes   data
+    );
+
+    event DelayedTransactionExecuted(
+        uint    id,
+        uint    timestamp,
+        uint    value,
+        bytes   data
+    );
+
+    address public ownedContract;
+
+    // Active list of delayed functions (delay > 0)
+    DelayedFunction[] public delayedFunctions;
+    // Map from function to the functions's location+1 in the `delayedFunctions` array.
+    mapping (bytes4 => uint) public delayedFunctionMap;
+
+    // Active list of delayed transactions
+    Transaction[] public delayedTransactions;
+    // Map from transaction ID to the transaction's location+1 in the `delayedTransactions` array.
+    mapping (uint => uint) public delayedTransactionMap;
+
+    // Used to generate a unique identifier for a delayed transaction
+    uint private totalNumDelayedTransactions = 0;
+
+    constructor(
+        address _ownedContract
+        )
+        public
+    {
+        require(_ownedContract != address(0), "ZERO_ADDRESS");
+        ownedContract = _ownedContract;
+    }
+
+    function executeDelayedTransaction(
```

Comment:
executeDelayedTransaction => executeDelayed ?
Make some methods shorter would be nice.